### PR TITLE
Expand resolved sample type values for CVR

### PR DESCRIPTION
**Pull request by ao508**
Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

### CODE REVIEW

File: cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java [hunk 1]
```diff
@@ -370,11 +370,20 @@ public void setMSK_SLIDE_ID(String mskSlideID) {
...
-        if (isMetastasis != null)
-            return isMetastasis == 0 ? "Primary" : "Metastasis";
+    public String resolveSampleType(Integer isMetastasis) {
+        if (isMetastasis != null) {
+            switch(isMetastasis) {
+                case 0:
+                    return "Primary";
+                case 1:
+                    return "Metastasis";
+                case 127:
```
Comment (sheridancbio) [inline cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java:380]: This is different from the description on the card. Local Recurrence had code of 2, and Unknown had a code of 127. Theoretically any other code coming in is an unexpected condition (maybe an exception or a reason to substitute NA)?

File: cvr/src/test/java/org/cbioportal/cmo/pipelines/cvr/CVRClinicalRecordTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        String sampleType = record.resolveSampleType(null);
+        Assert.assertTrue(sampleType.isEmpty());
+    }
+
+    @Test
+    public void testSampleTypeNotNull() throws Exception {
+        CVRClinicalRecord record = new CVRClinicalRecord();
+        String primarySampleType = record.resolveSampleType(0);
+        String metastasisSampleType = record.resolveSampleType(1);
+        String localRecurrenceSampleType = record.resolveSampleType(127);
```
Comment (sheridancbio) [inline cvr/src/test/java/org/cbioportal/cmo/pipelines/cvr/CVRClinicalRecordTest.java:61]: according to the card, sample type of 2 should resolve to localRecurrence

File: cvr/src/test/java/org/cbioportal/cmo/pipelines/cvr/CVRClinicalRecordTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        String defaultSampleType = record.resolveSampleType(128);
```
Comment (sheridancbio) [inline cvr/src/test/java/org/cbioportal/cmo/pipelines/cvr/CVRClinicalRecordTest.java:62]: according to the card, sample type 127 should resolve to the unknown sample type. Other codes should maybe resolve to null / empty / or cause an exception?  I'm not sure what is appropriate.

File: cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java [hunk 4]
```diff
@@ -370,11 +370,22 @@ public void setMSK_SLIDE_ID(String mskSlideID) {
         this.mskSlideID = mskSlideID;
     }
 
-    private String resolveSampleType(Integer isMetastasis) {
-        if (isMetastasis != null)
-            return isMetastasis == 0 ? "Primary" : "Metastasis";
+    public String resolveSampleType(Integer isMetastasis) {
```
Comment (sheridancbio) [inline cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java:373]: Sorry .. one more request ... the parameter to this function should probably be renamed to something like sampleType

Comment (ao508) [reply cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java:373]: note: we decided to keep the field as is to match the CVR JSON field that we get the value from ("is_metastasis")

Comment (sheridancbio) [reply cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/model/staging/CVRClinicalRecord.java:373]: after discussing, I understand that the DMP model still calls this field "isMetastasis", so the nomenclature carries through to this function call. Unless the DMP metadata model is updated (which maybe should be done) we can wait on making this change.

**Pull request merged**